test/jmap/core/session/session_test.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+
+void main() {
+  group('test get session', () {
+    test('session should correct from json', () {
```

Comment:
Add a test for the sessions in these files:

Handle missing tmail extension https://github.com/apache/james-project/blob/master/server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/SessionRoutesContract.scala

Add extra unknown properties (please don't hardcode them to pass the test)

https://github.com/apache/james-project/blob/master/server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/CustomMethodContract.scala

Line 80.

Bonus: handle also the session from fastmail folks.